android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java

```diff
@@ -65,6 +66,49 @@ public TiListView(ListViewProxy proxy)
 		this.recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
 		this.recyclerView.setFocusableInTouchMode(false);
 
+		// Add listener to fire scroll events.
+		this.recyclerView.addOnScrollListener(new RecyclerView.OnScrollListener()
+		{
+			@Override
+			public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState)
+			{
+				super.onScrollStateChanged(recyclerView, newState);
+
+				if (newState == RecyclerView.SCROLL_STATE_IDLE) {
+					proxy.fireSyncEvent(TiC.EVENT_SCROLLEND, generateScrollPayload());
+				}
+			}
+
+			@Override
+			public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy)
+			{
+				super.onScrolled(recyclerView, dx, dy);
+
+				proxy.fireSyncEvent(TiC.EVENT_SCROLLSTART, generateScrollPayload());
+			}
+		});
+		this.recyclerView.setOnFlingListener(new RecyclerView.OnFlingListener()
+		{
+			@Override
+			public boolean onFling(int velocityX, int velocityY)
+			{
+				final KrollDict payload = new KrollDict();
+
+				// Determine scroll direction.
+				if (velocityY > 0) {
+					payload.put(TiC.PROPERTY_DIRECTION, "down");
+				} else if (velocityY < 0) {
+					payload.put(TiC.PROPERTY_DIRECTION, "up");
+				}
+
+				// Set scroll velocity.
+				payload.put(TiC.EVENT_PROPERTY_VELOCITY, velocityY);
+
+				proxy.fireSyncEvent(TiC.EVENT_SCROLLING, payload);
+				return true;
```

Comment:
I can't scroll the ListView with a fling anymore. I think it's because you're returning `true` here. You should probably return `false` instead.